tests/transforms/test_defer_measurements.py

```diff
@@ -261,7 +263,7 @@ def qnode2(phi, theta):
 
         res2 = qnode2(np.pi / 4, 3 * np.pi / 4)
 
-        assert spy.call_count == 2
+        assert spy.call_count == 4
```

Comment:
why did this increase?

Reply:
I decorated tests of unsupported features (None shots or finite shots) with `qml.defer_measurements`. `qml.defer_measurements` used to be called twice during device initialization and it is now called two more times because of the decorators.

Reply:
ahh right, we still auto-call it on default.qubit in analytic mode. sounds good!